File: mobile/ios/FindlyKit/Sources/FindlyKit/Navigation/OnboardingVariant.swift
/// specs/010-app-shell-and-screen-ux.md §2.2 — the Onboarding screen's two variants. A
/// `PROFILE_NOT_FOUND` (no `Users` profile row at all, 001 §1.5.3) is materially different from a
/// `FAMILY_NOT_FOUND` (a profile exists but `familyId` is null — a groups-only user, 001 §1.5.4):
/// the profile-less caller cannot even call `GET /groups`, so only the four bootstrap paths make
/// sense for them, while the family-less caller already has a working Groups feature. Carried as
/// the `AppRoute.onboarding` associated value so the router — not the screen — is the single source
/// of truth for which variant is showing.
public enum OnboardingVariant: Equatable, Sendable {
    case profileLess
    case familyLess
}

File: mobile/android/app/src/main/java/com/findly/android/ui/onboarding/OnboardingVariant.kt
package com.findly.android.ui.onboarding

/**
 * The two Onboarding variants (specs/010-app-shell-and-screen-ux.md §2.2) — one route, two shapes,
 * replacing both places this UI lived before this spec (iOS Home's `profileless`/`familyless`
 * branches; Android's `GroupsListScreen` `ProfileNeeded` state — both retired).
 */
enum class OnboardingVariant {
    /** No `Users` profile row exists yet (001-api-contract.md §1.5.3, confirmed `PROFILE_NOT_FOUND`).
     * Welcome copy + a display-name field, then all four 001 §1.5.3 bootstrap paths. */
    ProfileLess,

    /** A profile exists but `familyId` is `null` (001 §1.5.4, confirmed `FAMILY_NOT_FOUND` on a
     * family-scoped load) — a groups-only user. No display-name field (they already have one). */
    FamilyLess,
}
